src/core/MIGRATION.md

```diff
@@ -642,22 +774,50 @@ With the legacy plugin system, extensions of core and other plugins are handled
 
 Each uiExport path is an entry file into one specific set of functionality provided by a client-side plugin. All webpack alias-based imports should be moved to these entry files, where they are appropriate. Moving a deeply nested webpack alias-based import in a plugin to one of the uiExport entry files might require some refactoring to ensure the dependency is now passed down to the appropriate place as function arguments instead of via import statements.
 
-### Switch to new platform services
+For the `plugins/` and `ui/` webpack aliases, you should be able to take advantage of the `legacy.ts` shim you created earlier. By placing these imports directly in your shim, you can pass the dependencies you need into your `Plugin.start` and `Plugin.setup` methods, from which point they can be passed down to the rest of your plugin's entry files.
+
+For items that don't yet have a clear "home" in the new platform, it may also be helpful to somehow indicate this in your shim to make it easier to remember that you'll need to change this later. One convention we've found helpful for this is simply using a namespace like `__LEGACY`:
+
+```ts
+// public/legacy.ts
+import { uiThing } from 'ui/thing';
+...
+
+const pluginInstance = plugin({} as PluginInitializerContext);
+const shimSetupPlugins = {
+  ...npSetup.plugins,
+  foo: fooSetup,
+  __LEGACY: {
+    uiThing, // eventually this will move out of __LEGACY and into a proper plugin
+  },
+};
+
+...
+export const setup = pluginInstance.setup(npSetup.core, shimSetupPlugins);
```

Comment:
One thing this doesn't show is how to shim **core** services that aren't migrated yet.

Also once #39945 is merged (happening today!), can we link to that table as well?